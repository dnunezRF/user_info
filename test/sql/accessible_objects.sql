\set ECHO 0
BEGIN;
\i sql/user_info.sql
\set ECHO all

-- create empty role
CREATE ROLE foo;

-- nothing yet owned by foo
SELECT type, relname, rights FROM accessible_objects('foo');

CREATE TABLE test_table (id INT);
GRANT ALL ON test_table TO foo;

SELECT type, relname, rights FROM accessible_objects('foo');

ROLLBACK;
