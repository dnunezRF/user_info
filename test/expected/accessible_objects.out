\set ECHO 0
-- create empty role
CREATE ROLE foo;
-- nothing yet owned by foo
SELECT type, relname, rights FROM accessible_objects('foo');
 type | relname | rights 
------+---------+--------
(0 rows)

CREATE TABLE test_table (id INT);
GRANT ALL ON test_table TO foo;
SELECT type, relname, rights FROM accessible_objects('foo');
   type   |  relname   |                          rights                           
----------+------------+-----------------------------------------------------------
 RELATION | test_table | {INSERT,SELECT,UPDATE,DELETE,TRUNCATE,REFERENCES,TRIGGER}
(1 row)

ROLLBACK;
